fix: revoke role on public for hooks (#27077)
supabase/supabase: apps/docs/content/guides/auth/auth-hooks.mdx

grant usage on schema public to supabase_auth_admin;
```

> Also revoke permissions from the `authenticated`, `public`, and `anon` roles to ensure the function is not accessible by Supabase Serverless APIs. We revoke access on the `public` role as it has access to functions created on `public.*` by default and `anon` and `authenticated` inherit permissions from the `public` role. Consequently, we need to revoke permissions from `public` in order to prevent the `anon` and `authenticated` roles from inheriting permissions to invoke the Postgres Hook.

Author: ```suggestion
Also revoke permissions from the `authenticated`, `public`, and `anon` roles to ensure the function is not accessible by Supabase Serverless APIs. The `public` role has  access to functions created on `public.*` by default and `anon` and `authenticated` inherit permissions from the `public` role.  Permission is revoked from the`public` role In order to prevent the `anon` and `authenticated` roles from inheriting permissions to invoke the Postgres Hook.
```

revoke execute
  on function public.custom_access_token_hook
  from authenticated, anon, public;

grant all
  on table public.profiles
  to supabase_auth_admin;

revoke all
  on table public.profiles

>   from authenticated, anon, public;

Author: There is no need to revoke from the tables on `public`. However we do this for consistency with function revocation.
